### PR TITLE
[12.0][FIX] intrastat_product: Excel report

### PR DESCRIPTION
**Pull request by manuelcalerosolis**
This PR is to protect the creation of excel reports when certain fields are left empty.

### CODE REVIEW

File: intrastat_product/report/intrastat_product_report_xls.py [hunk 1]
```diff
@@ -56,7 +53,7 @@ def _get_template(self, declaration):
                 'line': {
                     'type': 'string',
                     'value': self._render(
-                        "line.hs_code_id.local_code"),
+                        "line.hs_code_id.local_codeor or ''"),
```
Comment (pedrobaeza) [inline intrastat_product/report/intrastat_product_report_xls.py]: Incorrect

File: intrastat_product/report/intrastat_product_report_xls.py [hunk 2]
```diff
@@ -23,16 +20,16 @@ def _(self, src):
...
             'product': {
                 'header': {
                     'type': 'string',
                     'value': self._('Product'),
                 },
                 'line': {
+                    'type': 'string',
                     'value': self._render(
-                        "line.product_id and line.product_id.name"),
+                        "line.product_id and line.product_id.name or ''"),
```
Comment (pedrobaeza) [inline intrastat_product/report/intrastat_product_report_xls.py]:
```suggestion
                        "line.product_id.name or ''"),
```

File: intrastat_product/report/intrastat_product_report_xls.py [hunk 3]
```diff
@@ -274,4 +274,4 @@ def _intrastat_report(self, workbook, ws, ws_params, data, declaration):
             row_pos = self._write_line(
                 ws, row_pos, ws_params, col_specs_section='line',
                 render_space={'line': line},
-                default_format=self.format_tcell_left)
+                default_format=self.format_tcell_left)
```
Comment (pedrobaeza) [inline intrastat_product/report/intrastat_product_report_xls.py]: Missing ending extra line

**Comment by pedrobaeza** [comment]
Some comments pending

File: intrastat_product/report/intrastat_product_report_xls.py (hunk 2 above)
Comment (manuelcalerosolis) [reply intrastat_product/report/intrastat_product_report_xls.py]: corrected

File: intrastat_product/report/intrastat_product_report_xls.py (hunk 1 above)
Comment (manuelcalerosolis) [reply intrastat_product/report/intrastat_product_report_xls.py]: corrected

File: intrastat_product/report/intrastat_product_report_xls.py (hunk 3 above)
Comment (manuelcalerosolis) [reply intrastat_product/report/intrastat_product_report_xls.py]: corrected

**Review by pedrobaeza — APPROVED** [review]

**Comment by manuelcalerosolis** [comment]
Ready to be reviewed @ernestotejeda @chienandalu
Please review

**Comment by pedrobaeza** [comment]
I'm going to fast-track this as straight-forward and blocking for dependent modules:

/ocabot merge patch

**Pull request merged**